File: 00public_html/concentration/cpp/Card.cpp
#include "Card.hpp"
#include <algorithm>

Card::Card() {
  cardNum = -1;
  faceValue = -1;
  suit = "Null";
  name = "Null";
  picture = "Null";
}

Card::Card( int val ){
  cardNum = val;
  faceValue = val % 13 + 1;
  setSuit();
  setName();
  setPicture();
}

void Card::setSuit() {
  if( cardNum < 13 ) {
    suit = "Hearts";
  } else if( cardNum < 26 ) {
    suit = "Diamonds";
  } else if( cardNum < 39 ) {
    suit = "Clubs";
  } else {
    suit = "Spades";
  }
}

void Card::setName(){
  switch( cardNum % 13 ) {
    case 12: name = "King"; break;
    case 11: name = "Queen"; break;
    case 10: name = "Jack"; break;
    case 9: name = "Ten"; break;
    case 8: name = "Nine"; break;
    case 7: name = "Eight"; break;
    case 6: name = "Seven"; break;
    case 5: name = "Six"; break;
    case 4: name = "Five"; break;
    case 3: name = "Four"; break;
    case 2: name = "Three"; break;
    case 1: name = "Two"; break;
    case 0: name = "Ace"; break;
    default: name = to_string(cardNum % 13 + 1); break;
  }
}


char helpLower(char c) {
  return std::tolower(c);
}

void Card::setPicture() {
  string n = name;
  transform(n.begin(), n.end(), n.begin(), &helpLower );
  string s = suit;

  //using a lambda function to convert to lowercase
  //[] capture clause blank means we are not using any variables from the surrounding scope
  //-> char means the return type of the lambda function is char
  //transform takes a range and applies a function to each element in the range
  transform(s.begin(), s.end(), s.begin(), [](char c) -> char { return std::tolower(c); });
  picture = "cards/" + n + s + ".jpg";
}

string Card::getSuit() const {
  return suit;
}
string Card::getName() const {
  return name;
}

string Card::getPicture() const {
  return picture;
}

string Card::toString() const {
  return name + " of " + suit;
}

string Card::toHtml() const {
  stringstream ss;
  ss << "<div class=\"card\">"
     << "<img src=\"" << picture << "\" alt=\"" << toString() << "\">"
     << "<p>" << toString() << "</p>"
     << "</div>";
  return ss.str();
}

File: concentration/Card.js
class Card {
  constructor( cardNum) {
    this.cardNum = cardNum
    this.faceValue = cardNum % 13 + 1;
    this.setSuit();
    this.setName();
    this.setPicture();
  }

  setSuit() {
    if (this.cardNum < 13) {
      this.suit = "Hearts";
    } else if (this.cardNum < 26) {
      this.suit = "Diamonds";
    } else if (this.cardNum < 39) {
      this.suit = "Clubs";
    } else {
      this.suit = "Spades";
    }
  }

  setName() {
    switch (this.cardNum % 13) {
      case 12:
        this.name = "King";
        break;
      case 11:
        this.name = "Queen";
        break;
      case 10:
        this.name = "Jack";
        break;
      case 9:
        this.name = "Ten";
        break;
      case 8:
        this.name = "Nine";
        break;
      case 7:
        this.name = "Eight";
        break;
      case 6:
        this.name = "Seven";
        break;
      case 5:
        this.name = "Six";
        break;
      case 4:
        this.name = "Five";
        break;
      case 3:
        this.name = "Four";
        break;
      case 2:
        this.name = "Three";
        break;
      case 1:
        this.name = "Two";
        break;
      case 0:
        this.name = "Ace";
        break;
    }
  }

  setPicture() {
    this.picture = `cards/${this.name.toLowerCase()}${this.suit.toLowerCase()}.jpg`;
  }

  getFaceValue() {
    return this.faceValue;
  }

  getSuit() {
    return this.suit;
  }

  getName() {
    return this.name;
  }

  getPicture() {
    return this.picture;
  }

  getCardNum() {
    return this.cardNum;
  }

  toString() {
    return `${this.name} of ${this.suit}`;
  }

  toHtml() {
    return `<div class='card'>
              <img src='${this.picture}' alt='${this.toString()}'>
              <p>${this.toString()}</p>
              <p>#${this.getCardNum()}</p>
              <p>faceVal: ${this.getFaceValue()}</p>
              <p>Name: ${this.getName()}</p>
            </div>`;
  }
}
